modules/frontend/deduper_test.go

```diff
@@ -0,0 +1,117 @@
+package frontend
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/grafana/tempo/pkg/tempopb"
+	v1 "github.com/grafana/tempo/pkg/tempopb/trace/v1"
+	"github.com/grafana/tempo/pkg/util/test"
+)
+
+func TestDedupeSpanIDs(t *testing.T) {
+	tests := []struct {
+		name        string
+		trace       *tempopb.Trace
+		checkParent bool
+	}{
+		{
+			name: "no duplicates",
+			trace: &tempopb.Trace{
+				Batches: []*v1.ResourceSpans{
+					{
+						InstrumentationLibrarySpans: []*v1.InstrumentationLibrarySpans{
```

Comment:
would be better to have the input spans and expected output in each test instead of two tests that happen to result in the same output.